src/components/Steps/TestDetails.tsx

```diff
@@ -1,117 +1,119 @@
-import React, { useState } from "react";
 import styles from "../../styles/Home.module.css";
 
-import Select from "react-select";
+import { Step } from "@/pages/SessionCreator";
+import { MyForm } from "@/types/FormTypes";
+import { ActiveFormProps } from "@/types/types";
+import { SubmitHandler, useForm } from "react-hook-form";
 import {
   MarkingSchemeOptions,
+  OptionalLimitOptions,
   TestFormatOptions,
   TestPlatformOptions,
   TestPurposeOptions,
   TestTypeOptions,
-  OptionalLimitOptions,
 } from "../Options/TestDetailsOptions";
+import SelectField from "./Form/SelectedField";
 
 // Renders sub-page containing test details
 
-export default function TestDetails({ setActiveStep }) {
-  const [selectedTestType, setSelectedTestType] = useState(null);
-  const [selectedTestFormat, setSelectedTestFormat] = useState(null);
-  const [selectedTestPurpose, setSelectedTestPurpose] = useState(null);
-  const [selectedTestPlatform, setSelectedTestPlatform] = useState(null);
-  const [selectedMarkingScheme, setSelectedMarkingScheme] = useState(null);
-  const [selectedOptionalLimit, setSelectedOptionalLimit] = useState(null);
+export function TestDetails({ data, setActiveStep, setData }: ActiveFormProps) {
+  const { register, handleSubmit, control } = useForm<MyForm>({
+    defaultValues: { ...data.test },
```

Comment:
what is data.test

Reply:
It is the sample data being used as of now , that will be replaced by json in the next PR